src/shape/Symbols.tsx

```diff
@@ -86,16 +83,27 @@ const registerSymbol = (key: string, factory: D3SymbolType) => {
 
 export const Symbols = ({ type = 'circle', size = 64, sizeType = 'area', ...rest }: SymbolsProps) => {
   const props = { ...rest, type, size, sizeType };
+  let realType: SymbolType = 'circle';
+  if (typeof type === 'string') {
+    /*
+     * Our type guard is not as strong as it could be (i.e. non-existent),
+     * and so despite the typescript type saying that `type` is a `SymbolType`,
+     * we can get numbers or really anything, so let's have a runtime check here to fix the exception.
+     *
+     * https://github.com/recharts/recharts/issues/6197
+     */
+    realType = type;
```

Comment:
I think the spread logic outside of this component probably needs changed eventually too since that's the root reason this happens